azkaban-web-server/src/main/java/azkaban/webapp/TriggerPluginLoader.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.webapp;
+
+import static azkaban.ServiceProvider.SERVICE_PROVIDER;
+
+import azkaban.utils.FileIOUtils;
+import azkaban.utils.Props;
+import azkaban.utils.PropsUtils;
+import azkaban.webapp.plugin.TriggerPlugin;
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.apache.velocity.app.VelocityEngine;
+import org.mortbay.jetty.servlet.Context;
+
+public class TriggerPluginLoader {
+
+  private static final Logger log = Logger.getLogger(TriggerPluginLoader.class);
+  private final String pluginPath;
+
+  public TriggerPluginLoader(final Props props) {
```

Comment:
in the web server code. The `loadTriggerPlugins` method was extracted into a separate class. This requires further refactoring but phasing it for now.